jetstream/cli.py

```diff
@@ -182,12 +179,56 @@ def export_statistics_to_json(project_id, dataset_id, bucket):
 @cli.command("rerun_config_changed")
 @project_id_option
 @dataset_id_option
-@dry_run_option
-def rerun_config_changed(project_id, dataset_id, dry_run):
+@click.pass_context
+def rerun_config_changed(ctx, project_id, dataset_id):
     """Rerun all available analyses for experiments with new or updated config files."""
     # get experiment-specific external configs
     external_configs = ExternalConfigCollection.from_github_repo()
 
     updated_external_configs = external_configs.updated_configs(project_id, dataset_id)
     for external_config in updated_external_configs:
-        rerun(project_id, dataset_id, external_config.experimenter_slug, dry_run)
+        ctx.invoke(rerun, project_id, dataset_id, external_config.experimenter_slug, dry_run=False)
+
+
+@cli.command("validate_config")
+@click.argument("path", type=click.Path(exists=True), nargs=-1)
+@click.pass_context
+def validate_config(ctx, path):
+    """Validate config files."""
+    config_files = [p for p in path if os.path.isfile(p)]
+
+    collection = ExperimentCollection.from_experimenter()
+
+    # required to resolve the config
+    dummy_experiment = Experiment(
+        slug="config_dummy",
+        type="pref",
+        status="Complete",
+        start_date=dt.datetime(2020, 3, 1, tzinfo=pytz.utc),
+        end_date=dt.datetime(2020, 3, 1, tzinfo=pytz.utc),
+        proposed_enrollment=1,
+        variants=[],
+    )
+
+    for file in config_files:
+        click.echo(f"Validate {file}", err=False)
+
+        spec = AnalysisSpec.from_dict(toml.load(file))
+        config = spec.resolve(dummy_experiment)
+
+        # check if there is an experiment with a matching slug in Experimenter
+        slug = os.path.splitext(os.path.basename(file))[0]
+        if collection.with_slug(slug).experiments == []:
+            click.echo(f"No experiment with slug {slug} in Experimenter.", err=False)
+        else:
+            # dry run experiment analysis with the config file
+            # this will make sure config file contents are valid
+            ctx.invoke(
+                run,
+                date=config.experiment.start_date,
+                experiment_slug=slug,
+                dry_run=True,
+                config_file=file,
+            )
```

Comment:
Currently skips dry run for experiments that do not exist. I guess we can create a dummy `Experiment` for this somewhere, but that'd require adding more complexity to `run`.